cmd/snap/cmd_model.go

```diff
@@ -0,0 +1,186 @@
+	Verbose   bool `long:"verbose"`
+	Assertion bool `long:"assertion"`
+}
+
+func init() {
+	addCommand("model",
+		shortModelHelp,
+		longModelHelp,
+		func() flags.Commander {
+			return &cmdModel{}
+		}, waitDescs.also(map[string]string{
+			"assertion": i18n.G("Print the raw assertion."),
+			"verbose":   i18n.G("Print more of the assertion fields."),
+			"serial":    i18n.G("Print the serial assertion instead of the model."),
+		}),
+		[]argDesc{},
+	)
+}
+
+func (x *cmdModel) Execute(args []string) error {
+	var assertion asserts.Assertion
+	var err error
+	if x.Serial {
+		assertion, err = x.client.CurrentSerialAssertion()
+	} else {
+		assertion, err = x.client.CurrentModelAssertion()
+	}
+
+	if err != nil {
+		return err
+	}
+
+	// --assertion means output the raw assertion
+	if x.Assertion {
+		_, err = Stdout.Write(asserts.Encode(assertion))
+		return err
+	}
+
+	termWidth, _ := termSize()
+	termWidth -= 3
+	if termWidth > 100 {
+		// any wider than this and it gets hard to read
+		termWidth = 100
+	}
+
+	w := tabWriter()
+
+	primaryHeaderKeys := assertion.Type().PrimaryKey
+	for _, header := range primaryHeaderKeys {
+		val := assertion.HeaderString(header)
+		err = maybeWrapSingleYAMLKey(w, header, val, termWidth)
+		if err != nil {
+			return err
+		}
+	}
+
+	// --verbose means output all of the fields
+	if x.Verbose {
+		allHeadersMap := assertion.Headers()
+		allHeaders := make([]string, len(allHeadersMap))
+		i := 0
+		for k := range allHeadersMap {
+			allHeaders[i] = k
+			i++
+		}
+
+		sort.Strings(allHeaders)
+		for _, headerName := range allHeaders {
+			header := allHeadersMap[headerName]
+			// don't print any meta headers except revision, which is important
+			// for a device which has remodeled
+			if headerName != "revision" && stringInList(headerName, asserts.MetaHeaders) {
+				continue
+			}
+			if stringInList(headerName, primaryHeaderKeys) {
+				continue
+			}
+
+			switch v := header.(type) {
+			case int, uint, uint64, int64, float32, float64, string, bool:
+				// handle simple scalar values with go's default format specifier
+				err = maybeWrapSingleYAMLKey(w, headerName, fmt.Sprint(v), termWidth)
+				if err != nil {
+					return err
+				}
+			default:
+				// marshal the value as yaml, then print each line with wrapLine
+				bytes, err := yaml.Marshal(header)
+				if err != nil {
+					return err
+				}
+				_, err = fmt.Fprintf(w, "%s:\t\n", headerName)
+				if err != nil {
+					return err
+				}
+				for _, line := range strings.Split(string(bytes), "\n") {
+					if len(strings.TrimSpace(line)) == 0 {
+						continue
+					}
+					err = wrapLine(w, []rune(line), "  ", termWidth)
+					if err != nil {
+						return err
+					}
+				}
+			}
+		}
+	}
+
+	return w.Flush()
+}
+
+func stringInList(s string, list []string) bool {
+	for _, elem := range list {
+		if elem == s {
+			return true
+		}
+	}
+	return false
+}
+
+func maybeWrapSingleYAMLKey(w io.Writer, key, val string, termWidth int) error {
```

Comment:
This has the problem of wrapping over-aggressively when things are already multi-line, as well as getting things wrong by counting `\t` as one character when it could be multiple.

I think with a little bit of knowledge of what exactly it's printing things can be made a lot better and simpler, and I don't think that knowledge is a bad thing. As this is ad-hoc for models and serials we can use this specificity. IMHO; maybe @pedronis knows more.